js/lex.go

```diff
@@ -66,21 +90,34 @@ func (tt TokenType) String() string {
 
 // Lexer is the state for the lexer.
 type Lexer struct {
-	r *buffer.Lexer
-
-	regexpState   bool
-	templateState bool
+	r     *buffer.Lexer
+	stack []ParsingContext
```

Comment:
Ah. Well it's "parsing context stack", just "context" would be incorrect because context is just item of it (as in current context). Maybe `contextStack` but that's slightly long...